js/index.js

```diff
@@ -55,16 +58,188 @@ function setActive(e) {
-navigationLinks.forEach((link) => {
-  link.addEventListener("click", setActive);
-});
+
+function removeMessage(e){
+    //Find the button's parent element using DOM:
+    const entry = e.target.parentElement;
+    //Remove the entry element from the DOM:
+    entry.remove();
+    //Set the messages section visibility:
+    setMessagesVisibility();
+}
+
+function showModal(e){
+   //Get the modal container
+   const modalContainer = document.querySelector(".modal-container");
+   //Close the modal container
+   modalContainer.classList.add("show"); 
+}
+
+function closeModal(e){
+  //Get the modal container
+  const modalContainer = document.querySelector(".modal-container");
+  //Close the modal container
+  modalContainer.classList.remove("show");
+}
+
+function editMessage(e){
+  //Getting the message elements:
+  const messageId = e.target.getAttribute("data-messageId");
+  const name = e.target.getAttribute("data-name");
+  const email = e.target.getAttribute("data-email");
+  const message = e.target.getAttribute("data-message");
+  console.log(`${messageId} + ${name} + ${email} + ${message} `);
+  //Getting the modal elements
+  const modalContainer = document.querySelector(".modal-container");
+  const editMessageForm = modalContainer.querySelector("#editMessageForm");
+  const closeButton = modalContainer.querySelector("#closeButton");
+  //Show the edit modal
+  showModal();
+  //Reset the edit modal form elements
+  editMessageForm.reset();
+  //Get the edit form modal HTML elements
+  const editName = editMessageForm.querySelector("#editName");
+  const editEmail = editMessageForm.querySelector("#editEmail");
+  const editMessage = editMessageForm.querySelector("#editMessage");
+  //Set the values of the edit form modal to the to-edit message values
+  editName.value = name;
+  editEmail.value = email;
+  editMessage.value = message;
+  //Add event listener to close button
+  closeButton.addEventListener("click", closeModal);
+  //Assign message id to editSubmit button
+  const editSubmitButton = editMessageForm.querySelector("#editSubmit");
+  editSubmitButton.setAttribute("data-messageId", messageId);
+}
+
+function handleMessageForm(e){
+  //Prevent default
+  e.preventDefault();
+  //Retrieve form's controls values
+  const name = e.target.name.value;
+  const email = e.target.email.value;
+  const message = e.target.message.value;
+  //Log the form's controls values
+  console.log(`Name: ${name} | Email: ${email} | message: ${message}`);
+  //Selecting messages section
+  const messageSection = document.querySelector("#messages");
+  //Selecting the message list inside the message section
+  const messageList = messageSection.querySelector("ul");
+  //Create a new list item
+  const newMessage = document.createElement("li"); 
+  //Set the innerHTML of the newMessage element
+  newMessage.innerHTML = `<input type="hidden" value="${messageId}"><a href="mailto:${email}">${name}</a>
+  <textarea>${message}</textarea>`;
```

Comment:
This should be in a `span`.

Reply:
Thank you for your feedback. I agree, I missed the part where it required us to display the message using `span`. I will fix it on the next assignment!